### PR TITLE
Add and ship file that lists the SDK versions XI/XM supports.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Add and ship a file that lists the SDK versions of the various SDKs XI supports.

Also list the minimum SDK version for each extension.

And add a script that verifies that the current SDK version is in this file,
which should ensure every SDK version bump ends up in the file.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
@VincentDondain @jstedfast @emaf I believe this is useful for you. Please let me know if you want a different format or more data.

File: tools/mtouch/SdkVersions.plist [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			<string>10.1</string>
+		</array>
+		<key>tvOS</key>
+		<array>
+			<string>9.0</string>
+			<string>9.1</string>
+			<string>9.2</string>
+			<string>9.3</string>
+			<string>10.0</string>
+			<string>10.1</string>
```
Comment (spouliot) [inline tools/mtouch/SdkVersions.plist]: tvOS 10.1 is part of Xcode 8.2

File: tools/mtouch/SdkVersions.plist [hunk 2]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (spouliot) [inline tools/mtouch/SdkVersions.plist]: should not this file one generated (and not checked in) ?

File: tools/mtouch/SdkVersions.plist.in [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			<string>10.1</string>
+		</array>
+		<key>tvOS</key>
+		<array>
+			<string>9.0</string>
+			<string>9.1</string>
+			<string>9.2</string>
+			<string>9.3</string>
+			<string>10.0</string>
+			<string>10.1</string>
```
Comment (spouliot) [inline tools/mtouch/SdkVersions.plist.in]: same (or better here)

File: tools/mtouch/SdkVersions-check.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+var failed = false;
+if (doc.SelectSingleNode ($"/plist/dict/dict/key[text()='iOS']/following-sibling::array/string[text()='{args [2]}']") == null) {
+	Console.Error.WriteLine ($"Could not find the iOS version {args [2]} in SdkVersions.plist.in.");
+	failed = true;
+}
+
+if (doc.SelectSingleNode ($"/plist/dict/dict/key[text()='tvOS']/following-sibling::array/string[text()='{args [3]}']") == null) {
+	Console.Error.WriteLine ($"Could not find the tvOS version {args [3]} in SdkVersions.plist.in.");
+	failed = true;
```
Comment (spouliot) [inline tools/mtouch/SdkVersions-check.cs]: that failed to detect the tvOS 10.1 when the maximum is 10.0, OTOH that should not be the most common mistake

File: tools/mtouch/SdkVersions.plist.in [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		</array>
+		<key>tvOS</key>
+		<array>
+			<string>9.0</string>
+			<string>9.1</string>
+			<string>9.2</string>
+			<string>9.3</string>
```
Comment (spouliot) [inline tools/mtouch/SdkVersions.plist.in]: IIRC there was no 9.3 since tvOS lagged by 0.1 with iOS version numbers last year

**Comment by spouliot** [comment]
@chamons XM should have something similar so your feedback is welcome too.

**Comment by chamons** [comment]
@spouliot I agree. That first entails XM getting our act together and officially determining what SDKs we test with / claim to support.

It'll take at least a bit of research. Filed card here so I don't forget -

https://trello.com/c/Q9Y0WHwR/309-figure-out-sdk-version-question-and-add-to-plist-file

File: tools/mtouch/SdkVersions.plist.in (hunk 3 above)
Comment (rolfbjarne) [inline tools/mtouch/SdkVersions.plist.in]: master supports iOS 10.1 (but not 10.2).

File: tools/mtouch/SdkVersions.plist (hunk 2 above)
Comment (rolfbjarne) [inline tools/mtouch/SdkVersions.plist]: fixed

File: tools/mtouch/SdkVersions.plist (hunk 1 above)
Comment (rolfbjarne) [inline tools/mtouch/SdkVersions.plist]: fixed

File: tools/mtouch/SdkVersions.plist.in (hunk 5 above)
Comment (rolfbjarne) [inline tools/mtouch/SdkVersions.plist.in]: fixed

**Comment by rolfbjarne** [comment]
@spouliot it's not really a list of supported extensions, it's a list of min version for each extension (so that the IDE can compute the correct list of min deployment targets for those extensions).

**Comment by jstedfast** [comment]
@rolf awesome! The information in the SdkVersions.plist looks fantastic. I cannot think of anything else we need.

**Comment by chamons** [comment]
Ok, since it's just a list of os's we support, XM would be 10.7 -> 10.12 right now.

Talking to @rolfbjarne on Slack makes it sounds like we'll need a separate file in  /Library/Frameworks/Xamarin.Mac.framework for the mac bits.

**Comment by VincentDondain** [comment]
To add on Sebastien's previous comment:

> I had not thought about having the list of supported extensions. While it's nice to have the information handy I wonder how that can be used, as the templates must already have a similar logic (I assume) and they are not shipped with XI.

While the info *seems handy*, I don't see right now the use of the extensions min versions. The iOS extension wizard actually doesn't **yet** let people select a `target minimum iOS version` but it's gonna change soon. At that time, what we'll likely do (same as iOS templates) is have each extension template define its own minimum iOS version.

**Comment by rolfbjarne** [comment]
> Though, it likely doesn't hurt to have it, maybe there's a usage I can't think of right now.

I was thinking about the Plist editor, where you get a list of the valid minimum deployment targets.

**Comment by VincentDondain** [comment]
@rolfbjarne yes excellent point that would totally work and we absolutely don't do that right now.

**Comment by spouliot** [comment]
@rolfbjarne lgtm 👍

**Pull request merged**